### PR TITLE
Fix useAuthState hook docs

### PR DESCRIPTION
**Pull request by WiXSL**

### CODE REVIEW

File: packages/ra-core/src/auth/useAuthState.ts [hunk 1]
```diff
@@ -12,20 +12,15 @@ interface State {
...
  *
- * The return value updates according to the authProvider request state:
- *
- * - start:   { authenticated: false, loading: true, loaded: false }
- * - success: { authenticated: true,  loading: false, loaded: true }
- * - error:   { authenticated: false, loading: false, loaded: true }
- *
- * Useful in custom page components that can work both for connected and
- * anonymous users. For pages that can only work for connected users,
- * prefer the useAuthenticated() hook.
```
Comment (djhi) [inline packages/ra-core/src/auth/useAuthState.ts:28]: I think we should keep the initial format:

> Hook for ...
> Calls the dataProvider
> State changes
> Why and when to use it

**Review by djhi — CHANGES_REQUESTED** [review]

File: packages/ra-core/src/auth/useAuthState.ts (hunk 1 above)
Comment (WiXSL) [reply packages/ra-core/src/auth/useAuthState.ts]: Done. I hope is ok. Let me know

**Comment by WiXSL** [comment]
I've added a fix for a typo in the JDoc of useCheckAuth as well. I hope is ok

**Review by djhi — CHANGES_REQUESTED** [review]
:ok_hand: Thanks!

**Review by djhi — APPROVED** [review]

**Pull request merged**